documentation/terraform.md

```diff
@@ -0,0 +1,24 @@
+### Creating a new Terraform environment
+If we want to create a new long term environment, we need to create a new backing store for state.
+
+We are using S3 buckets created through GovPaaS & Cloudfoundry to store terraform state.
+In order to create a new bucket, you need to have the cloudfoundry cli installed and logged in to the GovPaaS account.
+
+To create a new S3 bucket, run
+
+```cf create-service aws-s3-bucket default dfe-ecf-terraform-state-<env>```
+
+To create the access key, run
+
+```cf create-service-key dfe-ecf-terraform-state-<env> terraform-state-key-<env> -c '{"allow_external_access": true}'```
+
+To view the access key:
+
+```cf service-key dfe-ecf-terraform-state-<env> terraform-state-key-<env>```
+
+You should be able to see the bucket_name, aws_access_key_id and aws_secret_access_key. With these values, run
+
+```terraform init -backend-config="bucket=<bucket_name>" -backend-config="access_key=<aws_access_key_id>" -backend-config="secret_key=<aws_secret_access_key>"```
```

Comment:
Probably worth saying how to set up terraform in the first place. Say, by giving a link to their official site.

Reply:
👍  will do